cmd/server/server.go

```
@@ -184,7 +185,11 @@ func (s *server) startService() common.Daemon {
 	params.PublicClient = workflowserviceclient.New(dispatcher.ClientConfig(common.FrontendServiceName))
 
 	if params.ClusterMetadata.ArchivalConfig().ConfiguredForArchival() {
-		params.BlobstoreClient, err = filestore.NewClient(&s.cfg.Archival.Filestore)
+		if s.cfg.Archival.Filestore != nil {
```

Can we return an error if both configs are specified.